### PR TITLE
[Helm]feat: add support for reverse proxy through extraContent

### PR DESCRIPTION
**Pull request by karimatwa**

### CODE REVIEW

File: helm-chart/values.yaml [hunk 1]
```diff
@@ -49,7 +49,9 @@ controlPlane:
       cpu: "1"
   # Security context for the pod or container
   securityContext: {}
-  #extra_content: {}  # Extra content (currently commented out)
+  #extraContent: # Extra content, needed if you want to configure a reverse proxy for control plane.
```
Comment (tpetillot) [inline helm-chart/values.yaml:52]: question: why in extraContent ?

Comment (karimatwa) [reply helm-chart/values.yaml:52]: To dynamically inject new configuration without needing to update the control plane configuration on Helm every time, it minimizes the need for frequent changes. WDYT?

Comment (tpetillot) [reply helm-chart/values.yaml:52]: But don't we lost helm generated documentation?
I see no difference with adding the field for real

File: helm-chart/README.md [hunk 2]
```diff
@@ -39,6 +39,9 @@ helm show values gatling/enterprise-locations-packages > values.yaml
 > [!TIP]
 > When connecting to the cluster using HTTPS, if a custom truststore and/or keystore is needed, `KUBERNETES_TRUSTSTORE_FILE`, `KUBERNETES_TRUSTSTORE_PASSPHRASE` and/or `KUBERNETES_KEYSTORE_FILE`, `KUBERNETES_KEYSTORE_PASSPHRASE` environment variables should be set.
 
+> [!TIP]
+> The control plane and location configurations allow for the setup of a reverse proxy. To enable this, uncomment the extraContent section and its associated values in the configuration file for the control plane and/or location. Ensure the reverse proxy is configured to rewrite the Host header to `api.gatling.io` to ensure proper functionality.
```
Comment (slandelle) [inline helm-chart/README.md:43]: Actually, the correct term is "forward proxy"

File: helm-chart/values.yaml (hunk 1 above)
Comment (slandelle) [reply helm-chart/values.yaml:52]: ~~reverse~~ => forward

File: helm-chart/values.yaml [hunk 3]
```diff
@@ -49,7 +49,9 @@ controlPlane:
...
+    #enterprise-cloud:
+      #url: "http://private-control-plane-reverse-proxy/gatling"
```
Comment (slandelle) [inline helm-chart/values.yaml:54]: ~~reverse~~=> forward

**Review by slandelle — CHANGES_REQUESTED** [review]

File: helm-chart/README.md (hunk 2 above)
Comment (karimatwa) [reply helm-chart/README.md]: Thanks for forwarding!

File: helm-chart/values.yaml (hunk 1 above)
Comment (karimatwa) [reply helm-chart/values.yaml]: Updated and fixed.

File: helm-chart/README.md [hunk 4]
```diff
@@ -39,6 +39,9 @@ helm show values gatling/enterprise-locations-packages > values.yaml
 > [!TIP]
 > When connecting to the cluster using HTTPS, if a custom truststore and/or keystore is needed, `KUBERNETES_TRUSTSTORE_FILE`, `KUBERNETES_TRUSTSTORE_PASSPHRASE` and/or `KUBERNETES_KEYSTORE_FILE`, `KUBERNETES_KEYSTORE_PASSPHRASE` environment variables should be set.
 
+> [!TIP]
+> The control plane and location configurations allow for the setup of a forward proxy. To enable this, uncomment the extraContent section and its associated values in the configuration file for the control plane and/or location. Ensure the forward proxy is configured to rewrite the Host header to `api.gatling.io` to ensure proper functionality.
```
Comment (tpetillot) [inline helm-chart/README.md:43]: It's not in extraContent anymore 😁

File: helm-chart/templates/_config.tpl [hunk 5]
```diff
@@ -31,7 +49,12 @@ control-plane {
         }
       }
       {{- end }}
-      debug.keep-load-generator-alive = {{ toJson .keepLoadGeneratorAlive }}
+      {{- with .extraContent }}
+      {{- range $key, $value := . }}
+      {{ $key }} = {{ toJson $value }}
+      {{- end }}
+      {{- end }}
```
Comment (tpetillot) [inline helm-chart/templates/_config.tpl:56]: should we keep this?

**Review by tpetillot — CHANGES_REQUESTED** [review]

File: helm-chart/templates/_config.tpl (hunk 5 above)
Comment (karimatwa) [reply helm-chart/templates/_config.tpl:56]: I kept it here since you added it to AWS Terraform configs for dev purposes. Will you need it in the future?

**Pull request merged**